src/tiered/tiered_work.c

```diff
@@ -139,7 +150,7 @@ __wt_tiered_pop_work(
     *entryp = entry = NULL;
 
     conn = S2C(session);
-    if (TAILQ_EMPTY(&conn->tieredqh))
+    if (__tiered_queue_peek_empty(conn))
```

Comment:
`__tiered_queue_peek_empty` is accessed outside of the tiered_lock to try save us from taking the lock unnecessarily. It's unsafe since every other access is lock protected, but it's an optimisation for perf reasons that I'm happy to suppress.
Ideally I'd make a general `__queue_peek` function, but changing `TAILQ` code from macros to functions is _not_ looking easy.

Reply:
What you have now is fine.